recipes/make/all/conanfile.py

```diff
@@ -13,45 +16,42 @@ class MakeConan(ConanFile):
     license = "GPL-3.0-or-later"
     settings = "os", "arch", "compiler", "build_type"
 
-    exports_sources = "patches/*"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
 
     def package_id(self):
         del self.info.settings.compiler
 
-    def build(self):
-        for patch in self.conan_data.get("patches").get(self.version, []):
-            tools.patch(**patch)
+    def _patch_sources(self):
+        apply_conandata_patches(self)
 
-        with tools.chdir(self._source_subfolder):
-            # README.W32
-            if tools.os_info.is_windows:
-                if self.settings.compiler == "Visual Studio":
-                    command = "build_w32.bat --without-guile"
-                else:
-                    command = "build_w32.bat --without-guile gcc"
+    def build(self):
+        self._patch_sources()
+        # README.W32
+        if is_msvc(self):
+            if self.settings.compiler == "Visual Studio":
+                command = "build_w32.bat --without-guile"
             else:
-                env_build = AutoToolsBuildEnvironment(self)
-                env_build.configure()
-                command = "./build.sh"
-            with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
```

Comment:
How to rewrite this for conan 2.0?

Is there self.run argument, that adds vcvars?